modules/nextflow/src/main/groovy/nextflow/util/FileArchiver.groovy

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020-2022, Seqera Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package nextflow.util
+
+import java.nio.file.Path
+
+import groovy.transform.CompileStatic
+import nextflow.file.FileHelper
+
+/**
+ * Helper class to resolve bucket archive paths
+ *
+ * @author Paolo Di Tommaso <paolo.ditommaso@gmail.com>
+ */
+@CompileStatic
+@Singleton(lazy = true, strict = false)
+class FileArchiver {
+
+    private final Path baseDir
+    private final Path targetDir
+
+    Path getBaseDir() { return baseDir }
+
+    Path getTargetDir() { return targetDir }
+
+    boolean asBoolean() {
+        return baseDir!=null && targetDir!=null
+    }
+
+    FileArchiver() {
+        this(System.getenv())
+    }
+
+    FileArchiver(Map<String,String> env) {
+        final paths = env.get('NXF_ARCHIVE_DIR')?.tokenize(',')
```

Comment:
A `,` is a valid path character. I known that we'll have the same problem with any character but may be we can use two characters to do the splitting and reduce the probabilities that is part of the path. Also a `,` seems more appropriate to separate items of a list and this is not the case. For me `==` fits slightly better in this case.

Reply:
yeah, I thought that, was a bit lazy to escape. 

> I known that we'll have the same problem with any character

Maybe no! I was not aware that has separator specific codes  https://stackoverflow.com/a/18782271/395921